### PR TITLE
Convert java-http-client to use Instrumenter API

### PR DESCRIPTION
**Pull request by jack-berg**
Contributing to Issue #2713.

### CODE REVIEW

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpHeadersInjectAdapter.java [hunk 1]
```diff
@@ -5,15 +5,40 @@
...
 import io.opentelemetry.context.propagation.TextMapSetter;
+import java.net.http.HttpHeaders;
 import java.net.http.HttpRequest;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-/** Context propagation is implemented via {@link HttpHeadersInstrumentation}. */
+/** Context propagation is initiated in via {@link HttpHeadersInstrumentation}. */
```
Comment (trask) [inline instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpHeadersInjectAdapter.java:18]:
```suggestion
/** Context propagation is initiated in {@link HttpHeadersInstrumentation}. */
```

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpClientSingletons.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    JdkHttpNetAttributesExtractor netAttributesExtractor = new JdkHttpNetAttributesExtractor();
+
+    INSTRUMENTER =
+        Instrumenter.<HttpRequest, HttpResponse<?>>newBuilder(
+                GlobalOpenTelemetry.get(), "io.opentelemetry.java-http-client", spanNameExtractor)
+            .setSpanStatusExtractor(spanStatusExtractor)
+            .addAttributesExtractor(httpAttributesExtractor)
+            .addAttributesExtractor(netAttributesExtractor)
+            .addAttributesExtractor(PeerServiceAttributesExtractor.create(netAttributesExtractor))
+            .addRequestMetrics(HttpClientMetrics.get())
```
Comment (trask) [inline instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpClientSingletons.java:40]: more metrics 🎉

**Review by trask — APPROVED** [review]
thx!

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpAttributesExtractor.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    return null;
+  }
+
+  @Override
+  protected @Nullable Integer statusCode(HttpRequest httpRequest, HttpResponse<?> httpResponse) {
+    return httpResponse.statusCode();
+  }
+
+  @Override
+  protected @Nullable String flavor(
```
Comment (mateuszrzeszutek) [inline instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpAttributesExtractor.java:65]: Nit

```suggestion
  protected String flavor(
```

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpAttributesExtractor.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  @Override
+  protected @Nullable Long requestContentLengthUncompressed(
+      HttpRequest httpRequest, @Nullable HttpResponse<?> httpResponse) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable Integer statusCode(HttpRequest httpRequest, HttpResponse<?> httpResponse) {
```
Comment (mateuszrzeszutek) [inline instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpAttributesExtractor.java:60]:
```suggestion
  protected Integer statusCode(HttpRequest httpRequest, HttpResponse<?> httpResponse) {
```

File: instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpNetAttributesExtractor.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JdkHttpNetAttributesExtractor
+    extends NetAttributesExtractor<HttpRequest, HttpResponse<?>> {
+
+  private static final Logger logger = LoggerFactory.getLogger(JdkHttpNetAttributesExtractor.class);
+
+  @Override
+  public @Nullable String transport(HttpRequest httpRequest) {
```
Comment (mateuszrzeszutek) [inline instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/JdkHttpNetAttributesExtractor.java:22]:
```suggestion
  public String transport(HttpRequest httpRequest) {
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Pull request merged**